app/src/main/java/org/simple/clinic/editpatient/EditPatientScreen.kt

```diff
@@ -351,6 +355,14 @@ class EditPatientScreen : BaseScreen<
     }
 
     deletePatient.setOnClickListener { router.push(DeletePatientScreen.Key(screenKey.patient.uuid)) }
+
+    setFragmentResultListener(RequestToScanIdentifier) { requestKey, result ->
+      if (result !is Succeeded) return@setFragmentResultListener
+
+      if (requestKey is RequestToScanIdentifier) {
+        val scannedIdentifier = (result.result as ScanSimpleIdScreen.SendScannedIdentifier).identifier
```

Comment:
I think we should change the name of `RequestToScanIdentifier` to something more specific like `ScannedBPPassport` or `ScannedNHID`. As currently, we have created an implicit object which would then not be helpful in the case of NHID. So, let's rename it to `ScannedBPPassport` and while passing it in `openScanSimpleIdScreen` we should check if opened from is `ToAddBPPassport` and then set the request type as `ScannedBPPassport`. WDYT?

Reply:
Oh I'll rename it to `RequestToScanBpPassport`, makes more sense

Reply:
We can maybe rename it to `ScannedBPPassportResult` or something like that as this key for the result and not request. You can also look at `ContactPatientSheet` for reference

Reply:
It is requesting to scan bp passport, not the result

Reply:
No.. So we are actually using `pushExpectingResult` in which the result is of type `RequestToScanIdentifier` 

Even in the example, you mentioned in another comment. They have named it `BlankScannedQrCode`

Reply:
So we're pushing expecting result where the key says what's going to happen. The key does not have to describe the result, the key is to basically to identify what's happening. So I'll remove the request and result both and rename it to something like `ScanBpPassport`

Reply:
Just an FYI, we can also pass the identifier directly as a result. Since it's a parcelable it should be fine. We can then check the request type and then read the result directly as well.